src/SLCore.Listeners/Implementation/AnalysisListener.cs

```diff
@@ -29,18 +29,42 @@ namespace SonarLint.VisualStudio.SLCore.Listeners.Implementation;
 [PartCreationPolicy(CreationPolicy.Shared)]
 internal class AnalysisListener : IAnalysisListener
 {
+    private readonly IAnalysisService analysisService;
+    private readonly IRaiseIssueParamsToAnalysisIssueConverter raiseIssueParamsToAnalysisIssueConverter;
+
+    [ImportingConstructor]
+    public AnalysisListener(IAnalysisService analysisService, IRaiseIssueParamsToAnalysisIssueConverter raiseIssueParamsToAnalysisIssueConverter)
+    {
+        this.analysisService = analysisService;
+        this.raiseIssueParamsToAnalysisIssueConverter = raiseIssueParamsToAnalysisIssueConverter;
+    }
+
     public Task DidChangeAnalysisReadinessAsync(DidChangeAnalysisReadinessParams parameters)
     {
         return Task.CompletedTask;
     }
 
     public void RaiseIssues(RaiseIssuesParams parameters)
     {
-        throw new NotImplementedException();
+        if (parameters.analysisId.HasValue)
+        {
+            var filepath = parameters.issuesByFileUri.Single().Key.LocalPath;
+
+            var issues = GetSupportedLanguageIssues(raiseIssueParamsToAnalysisIssueConverter.GetAnalysisIssues(parameters));
+
+            if (issues.Any())
+            {
```

Comment:
Clear on final raise issues? Not sure how it will work with raiseIssues + raiseHotspots, tho.

Reply:
Sorry, not clear, AnalysisService.CancelForFile

Also, we might want to do this as well `analysisStatusNotifierFactory.Create(nameof(SLCoreAnalyzer), path).AnalysisFinished()`

Reply:
Maybe CancelForFile is not the best idea, but consider the notifier factory

Reply:
Analysis finished should not be called from this class but from analysis service. We might need to pass `isIntermediatePublication` though.

Reply:
Nah. AnalysisService is not aware of sloop analysis. This is finalizing that. However, we might be able to call it from Sloop analyzer if the response to AnalyzeAndTrack is consistent

Reply:
AnalyzeAndTrack response can come before RaiseIssues so we can't do it there.

Reply:
Yeah so logically it should be done here, as it's the 2nd part of the Sloop analyzer